.github/workflows/storybook-chromatic-main.yml

```
@@ -1,13 +1,29 @@
 name: QA
 
 on:
-  push:
+  pull_request:
     branches:
-      - "main"
+      - main
+    types: [closed]
 
 jobs:
+  if_merged:
```

Define permissions in the job
```suggestion
  if_merged:
    permissions:
      pull-requests: read
```